src/cloud_provider/aws/router.rs

```diff
@@ -488,9 +489,6 @@ impl Delete for Router {
 
     fn on_delete_error(&self, target: &DeploymentTarget) -> Result<(), EngineError> {
         warn!("AWS.router.on_delete_error() called for {}", self.name());
-
-        send_progress_on_long_task(self, crate::cloud_provider::service::Action::Delete, || {
-            delete_stateless_service(target, self, true)
-        })
+        delete_router(target, self, true)
```

Comment:
won't this make an infinite loop?

Reply:
nop, it is a bit silly how it done ATM.
The boolean parameter is a flag to tell if it is from an error or not.
We do the same things except we fetch info from k8s (events, logs, ....) to send them in the failure.

So it is doing

Router.on_delete() -> delete_router(target, self, false)
 =>  Failure
Router.on_delete_error()  -> delete(target, self, true)
 => Failure but with infos
Run fails